Applets/UI/MainScreen/CellView.swift

```diff
@@ -8,46 +8,60 @@
-                    if model.percentOfDone < 100 {
-                        HStack(alignment: .lastTextBaseline) {
-                            Text("\(model.percentOfDone)")
-                                .font(.system(size: 100).bold().weight(.black))
-                                .padding(-10)
-                            Text("%")
-                                .font(.system(size: 40).bold().weight(.black))
-                                .padding(5)
-                        }
-                        .frame(width: 370, height: 100, alignment: .trailing)
-                        .foregroundColor(.white.opacity(0.30))
-                    } else {
-                        Text("DONE")
-                            .font(.system(size: 120).bold().weight(.black))
-                            .foregroundColor(.white.opacity(0.30))
+        ZStack(alignment: .center) {
+            VStack(alignment: .leading) {
+                Text(model.title ?? "")
+                    .font(.system(size: 20).bold())
+                    .frame(maxWidth: .infinity, alignment: .topLeading)
+                    .padding(.leading, 15)
+                    .padding(.top, 25)
+                    .padding(1)
+                    .foregroundColor(model.isEnabled ? .black : .gray)
+                
+                Text(model.goalDescription ?? "")
+                    .frame(maxWidth: .infinity, alignment: .topLeading)
+                    .padding(.leading, 15)
+                    .foregroundColor(model.isEnabled ? .gray : .gray.opacity(0.50))
+                
+                if model.fractionOfDone < 1 {
+                    ZStack {
+                        Rectangle()
+                            .frame(width: 330, height: 3)
+                            .foregroundColor(.clear)
+                            .modifier(FilledCellModifier(percentOfDone: model.fractionOfDone))
+                            .position(x: 182, y: 20)
+                        
+                        Circle()
+                            .foregroundColor(Color("peach"))
+                        Circle()
+                            .foregroundColor(.white)
                     }
                 }
-
-                VStack(alignment: .leading) {
-                    Text(model.title ?? "")
-                        .font(.largeTitle.bold())
-                        .frame(maxWidth: .infinity, alignment: .leading)
-                        .padding(.leading, 15)
-                        .foregroundColor(model.isEnabled ? .black : .gray)
-                    Text(model.timeFrame ?? "")
-                        .font(.subheadline)
-                        .frame(maxWidth: .infinity, alignment: .leading)
-                        .padding(.leading, 15)
-                        .foregroundColor(model.isEnabled ? .black : .gray)
+                
+                HStack {
+                    HStack {
+                        Image(systemName: "checkmark.circle")
```

Comment:
Need to have number of tasks

Reply:
Yes, you need to have number of completed steps and overall number of steps if this goal is not done yet